### DISCUSSION TITLE
Authentication - Additional error hooks - Handle duplicate accounts

### DISCUSSION THREAD

**Discussion by gkielwasser** [directus/directus#10948] [Feature Requests]
There is no way to handle public registration errors.

One usecase is to be able to handle email uniqness.

To reproduce:
- an account X created with the default provider
- public registration is activated

A user try to authenticate on the account X using an other provider than the default. An UNKNOWN_EXCEPTION is thrown because the email has to be unique.

Here is more detail about what have been tried in this discussion https://github.com/directus/directus/discussions/9695

@rijkvanzanten suggested to add more core hooks in the auth provider to be able to handle such errors https://github.com/directus/directus/discussions/9695#discussioncomment-1868636

@aidenfoxx feel free to tell if I can give additional information

**Comment by aidenfoxx** [comment]
All you would need to do is wrap the `getUserId` function in a try/catch in the authentication service and trigger the error hook there, the same as how it is done for the `login` method: https://github.com/directus/directus/blob/bd3ed753bac8039856a5fd4e484f7f05155b2da1/api/src/services/authentication.ts#L150

**Reply by gkielwasser** [reply]
@aidenfoxx Thanks a lot for help

I have open this PR https://github.com/directus/directus/pull/10966 Feel free to tell me if I can improve anything,

An other alternative to my issue would be also to move the `filter('auth.login')` at the very beginning of the login method so it would allow some operations before the method actually starts. As it might be breaking it is just a thinking :) .

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
